src/peep/peep.c

```diff
@@ -3421,7 +3478,787 @@ static void peep_update_fixing(int steps, rct_peep* peep){
+		sprite_move(x, y, peep->z, (rct_sprite *) peep); // TODO: arguments
+		invalidate_sprite_2((rct_sprite *) peep);
+		return;
+	}
+
+	loc_6C1653(steps, peep, ride);
+}
+
+void loc_6C0FD3(int steps, rct_peep *peep, rct_ride *ride) {
+	peep->sprite_direction = peep->var_78 << 3;
+	scenario_rand();
+	peep->action = PEEP_ACTION_STAFF_FIX;
+	if (steps & 0x100000) {
+		peep->action = PEEP_ACTION_STAFF_FIX_2;
+	}
+
+	// loc_6C0FFB:
+	peep->action_sprite_image_offset = 0;
+	peep->action_frame = 0;
+	sub_693B58(peep);
+	invalidate_sprite_2((rct_sprite *) peep);
+
+	loc_6C100D(steps, peep, ride);
+}
+
+static void loc_6C100D(int steps, rct_peep *peep, rct_ride *ride) {
+	// eax: steps
+	// esi: peep
+	// edi: ride
+
+	if (peep->action == 0xFF) {
+		loc_6C1653(steps, peep, ride);
+		return;
+	}
+
+	peep_update_action(); // TODO: arguments
+
+	uint8 al = peep->action;
+	sint8 ah = 0x25;
+	if (al != 0x0F) {
+		ah = 0x50;
+	}
+
+	// loc_6C1027
+	if (peep->action_frame != ah) {
+		return;
+	}
+
+	uint32 ebx = ride->broken_vehicle;
+	uint16 bp = ride->vehicles[ebx];
+
+	if (bp == 0xFFFF) {
+		loc_6C1653(steps, peep, ride);
+		return;
+	}
+
+	rct_sprite *sprite = &g_sprite_list[bp];
+	ah = ride->broken_car;
+
+	// loc_6C1061
+	do {
+		ah--;
+		if (ah < 0) {
+			sprite->vehicle.update_flags = 0xFEFF;
+			return;;
+		}
+
+		sprite = &g_sprite_list[sprite->vehicle.next_vehicle_on_train];
+	} while (true);
+}
+
+void loc_6C107B(int steps, rct_peep *peep, rct_ride *ride) {
+	peep->sprite_direction = peep->var_78 << 3;
+	peep->action = 18;
+	peep->action_sprite_image_offset = 0;
+	peep->action_frame = 0;
+
+	sub_693B58(peep);
+	invalidate_sprite_2((rct_sprite *) peep);
+
+	loc_6C10A5(steps, peep, ride);
+}
+
+static void loc_6C10A5(int steps, rct_peep *peep, rct_ride *ride) {
+	// eax: steps
+	// esi: peep
+	// edi: ride
+
+	if (peep->action == 0xFF) {
+		loc_6C1653(steps, peep, ride);
+		return;
+	}
+
+	peep_update_action(); // TODO: arguments
+	if (peep->action_frame != 0x65) {
+		return;
+	}
+
+	// loc_6C10C5
+	uint32 ebx = ride->broken_vehicle;
+	uint16 bp = ride->vehicles[ebx];
+	rct_sprite *sprite = &g_sprite_list[bp];
+	sint8 ah = ride->broken_car;
+
+	// loc_6C10FA:
+	while (true) {
+		ah--;
+		if (ah < 0) {
+			// loc_6C110D:
+			sprite->vehicle.update_flags &= 0xFDFF;
+			return;
+		}
+
+		sprite = &g_sprite_list[sprite->vehicle.next_vehicle_on_train];
+	}
+}
+
+void loc_6C1114(int steps, rct_peep *peep, rct_ride *ride) {
+
+	if (ride_type_has_flag(ride->type, 0x8008)) {
+		loc_6C1653(steps, peep, ride);
+		return;
+	}
+
+	uint16 ax = ride->station_starts[peep->current_ride_station];
+	if (ax == 0xFFFF) {
+		loc_6C1653(steps, peep, ride);
+		return;
+	}
+
+
+	uint8 dl = ride->station_heights[peep->current_ride_station];
+
+
+	uint16 cx = 0;
+	uint8 cl = ax >> 8;
+	uint8 al = ax & 0xFF;
+
+	ax = al << 5;
+	cx = cl << 5;
+	uint32 ebp = cx;
+	ebp = (ebp << 8) | ax;
+	ebp = ebp >> 3;
+	rct_map_element *mapElement = TILE_MAP_ELEMENT_POINTER(ebp);
```

Comment:
`map_get_track_element_at` you are getting the first station track element of the ride.